### PR TITLE
Add conditonal mark to skip dualtor io tests on single tors

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
Some dualtor only test scripts would be executed in single tor and xfail
#### How did you do it?
Add conditonal mark to skip dualtor io tests on single tors
#### How did you verify/test it?
Verified on both dualtor and single tor

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -421,6 +421,12 @@ dualtor/test_tunnel_memory_leak.py::test_tunnel_memory_leak:
     conditions:
       - "https://github.com/sonic-net/sonic-mgmt/issues/11403 and 'dualtor-64' in topo_name"
 
+dualtor_io/test_link_failure.py:
```
Comment (lolyu) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:424]: There are other dualtor_io testcases, should we include them all here?
image

Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml]: Yes I think so, I added in latest commit and tested

**Review by lolyu — APPROVED** [review]
LGTM

File: .azure-pipelines/pr_test_scripts.yaml [hunk 2]
```diff
@@ -24,7 +24,6 @@ t0:
   - container_hardening/test_container_hardening.py
   - database/test_db_config.py
   - database/test_db_scripts.py
-  - dhcp_relay/test_dhcp_pkt_recv.py
```
Comment (wangxin) [inline .azure-pipelines/pr_test_scripts.yaml:27]: Why remove this script from the t0 scenario?
DHCP relay is a critical feature of T0.

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 3]
```diff
@@ -421,6 +421,12 @@ dualtor/test_tunnel_memory_leak.py::test_tunnel_memory_leak:
     conditions:
       - "https://github.com/sonic-net/sonic-mgmt/issues/11403 and 'dualtor-64' in topo_name"
 
+dualtor_io:
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:424]: 1. This change is not really related with the above change. Suggest to use separate PR.
2. Impact of this change is big. All scripts under `dualtor_io` will be affected. Wondering why we didn't notice this gap? Is this really a gap?

File: .azure-pipelines/pr_test_scripts.yaml (hunk 2 above)
Comment (xwjiang-ms) [reply .azure-pipelines/pr_test_scripts.yaml:27]: Because in this script, there is a mark pytest.mark.topology('mx'), so I suppose it should only be run on mx

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml (hunk 3 above)
Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:424]: It's really a gap because smoke test would run dualtor_io test, and xfail some case, sometimes caused the calculation of passing rate lower than expected, then fail smoke test

File: .azure-pipelines/pr_test_scripts.yaml (hunk 2 above)
Comment (xwjiang-ms) [reply .azure-pipelines/pr_test_scripts.yaml:27]: I will use another PR to fix this

**Review by wangxin — APPROVED** [review]
